dd-sdk-android/src/main/kotlin/com/datadog/android/rum/tracking/ActivityViewTrackingStrategy.kt

```diff
@@ -94,6 +94,28 @@ class ActivityViewTrackingStrategy @JvmOverloads constructor(
 
     // endregion
 
+    // region Object
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as ActivityViewTrackingStrategy
+
+        if (trackExtras != other.trackExtras) return false
+        if (componentPredicate != other.componentPredicate) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
```

Comment:
is there a reason why other classes of this PR have `javaClass.hashCode()` as body of that method, and this class takes a different approach?

Reply:
I think the reason why is that there's no other field in those classes to based your hashcode function on.